prog.go

```diff
@@ -498,6 +498,29 @@ func (p *Program) Close() error {
 	return p.fd.Close()
 }
 
+// Various options for Run'ing a Program
+type RunOptions struct {
+	// Program's data input. Required field.
+	Data []byte
+	// Program's data after Program has run. Caller must allocate. Optional field.
+	// If the data output from the kernel is larger than the provided buffer, Run will panic.
```

Comment:
I think we should keep this as an implementation detail and not document it. On newer kernels we will get ENOSPC, and on older kernels only programs that extend the buffer may trigger this.